### ISSUE TITLE
inpaint_global_harmonious preprocessor

### ISSUE THREAD

**Issue by axelblaze88** [Acly/comfyui-inpaint-nodes#75]
Do these nodes include any preprocessor like inpaint_global_harmonious from automatic1111?

![inpaint-global-harmonious-wmd](https://github.com/user-attachments/assets/bef5d639-8d70-49fd-9332-18c6f547128a)

That feature works wonders for image restoration and I need to switch to ComfyUI for more flexibility

**Comment by valery016** [comment]
It is indeed a very special preprocessor that can be used for a number of purposes beyond image restoration (where [SILVI2 ](https://civitai.com/articles/5834/silvi-v2-upscale-method-super-inteligence-large-vision-image) is the most brilliant example). I use it for my current artistic project as the only suitable method of the kind, will try to demonstrate that with examples on the main discussion page. According to my brief research, there is no node for, or any other code implementation of inpaint_global_harmonious preprocessor in Comfy, and SwarmUI doesn't support it either (the developer doesn't even know what it is).  Still, I would love to have support for inpaint_global_harmonious in Krita Ai plugin too, if possible.

**Comment by axelblaze88** [comment]
Yes, that's exactly what I was trying to do, implement SILVI in Comfy. Wanna add some intermediate processes to the workflow that'd be impossible with AUTO1111's rigidity. A real shame they don’t implement it in Comfy, honestly :(

**Comment by valery016** [comment]
I am getting very good results with Hyper-SDXL-8steps-CFG-lora in Forge / SDXL, equivalent or actually better than those with SILIVI.  So hang on, I intend to publish results here and on reddit SD sub, also on CivitAI (obviously).  

In the meantime, here is what I designed for Krita in this regard. Feel free to play with it, see if you can reproduce it and/or perhaps improve it even further.

{
    "name": "HyperScale (XL)",
    "version": 1,
    "sd_version": "auto",
    "sd_checkpoint": "RealVisXL_V4.0.safetensors",
    "loras": [
        {
            "name": "Hyper-SDXL-8steps-CFG-lora.safetensors",
            "strength": 0.85
        }
    ],
    "style_prompt": "best quality, highres",
    "negative_prompt": "bad quality, low resolution, blurry",
    "vae": "Checkpoint Default",
    "clip_skip": 1,
    "v_prediction_zsnr": false,
    "self_attention_guidance": false,
    "preferred_resolution": 0,
    "sampler": "Alternative - Euler A",
    "sampler_steps": 8,
    "cfg_scale": 2.0,
    "live_sampler": "Realtime - Hyper",
    "live_sampler_steps": 8,
    "live_cfg_scale": 3.5
}

In addition, create Unblur control layer with these settings before generating (Strength 1.0 (!)), setting in the screenshot below.
Also, since the method adds significant red color cast to the output, use the Luminosity blending mode for the generated later to get rid of it.

![Krita screenshot](https://github.com/user-attachments/assets/34c5e0ff-4f42-4ff7-b9c4-c78e768a2305)


Here's my best result with Krita for the SILVI test image:


 
![UPSCALED Krita RealVisXLv4 HyperScale CNet tile 0 76-10 1x 4xFFHQDAT 2x 16s colorfixed](https://github.com/user-attachments/assets/9d511ea5-a0c8-4d04-a5b7-98116a411933)
